03-rbac/proposal.md

```diff
@@ -0,0 +1,110 @@
+# Summary
+
+Introduces a proposal for RBAC. Specifically:
+
+- What roles do we need?
+- How would fly set-team work?
+- How do roles get persisted in the database?
+- How are roles handled by the API?
+
+
+# Proposal
+
+## What roles do we need?
+
+`global_admin` or `admin` - Granted if you are part of an admin team. Currently only the `main` team is an admin team.
+
+`team_member` or `member` - Granted during `fly set-team`. Has read/write access to all the team's resources.
+
+`team_viewer` or `viewer` - Granted during `fly set-team`. Has readonly access to all the team's resources.
+
+`public` - Not a role. This applies to pipelines. Public pipelines will behave as they did before.
+
+
+## How would fly set-team work?
+
+Currently you fly set-team like the following:
+```bash
+fly -t mytarget set-team -n myteam --allow-all-users
+fly -t mytarget set-team -n myotherteam --github-user pivotal-jwinters --github-team myorg:myteam
+```
+
+We could allow roles to be set as follows:
+```bash
+fly -t mytarget set-team -n myteam --role viewer --allow-all-users
+fly -t mytarget set-team -n myteam --role member --github-user pivotal-jwinters --github-team myorg:myteam
+```
+
+The main drawback here is that we would have to merge `auth` configuration because `viewers` and `members` would be configured in two separate steps.  The current behaviour overwrites the entire auth config each time, so in keeping with the current behaviour I guess we would overwrite the entire config for each role separately. 
```

Comment:
I think this makes sense, but at this point maybe we should explore having team config files instead, and having it set everything at once, including all the roles. Kind of like Vault's policy config files: https://www.vaultproject.io/docs/concepts/policies.html (in spirit)

We could then allow configuring multiple teams on ATC start by pointing it at the config files. This would even make the `main` team no longer "special" - it'd just be among the set of teams configured at start, with `admin` set to `true`.

Reply:
@xtreme-sameer-vohra I'm wondering if your exported team list should match the config format that we choose?

Reply:
@pivotal-jwinters Yep, it makes sense for `fly teams` to be able to export team contents in the same format such that it could be used for `set-teams` as well/